firestore/firestore_snippets/solution_counters.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strconv"
+
+	"cloud.google.com/go/firestore"
+	"google.golang.org/api/iterator"
+)
+
+// [START fs_counter_classes]
+// counters/${ID}
+type Counter struct {
+	numShards int
+}
+
+// counters/${ID}/shards/${NUM}
+type Shard struct {
+	Count int
+}
+
+// [END fs_counter_classes]
+
+// [START fs_create_counter]
+func (c *Counter) сreateCounter(ctx context.Context, docRef *firestore.DocumentRef) error {
+	// Initialize the counter document, then initialize each shard.
+	var err error
+	colRef := docRef.Collection("shards")
+
+	// Initialize each shard with count=0
+	for num := 0; num < c.numShards; num++ {
+		shard := Shard{0}
+
+		_, err = colRef.Doc(strconv.Itoa(num)).Set(ctx, shard)
```

Comment:
Please inline the error declaration. No need to pre-declare it above, either.
```suggestion
		if _, err := colRef.Doc(strconv.Itoa(num)).Set(ctx, shard); err != nil {
```